fdbserver/storageserver.actor.cpp

```diff
@@ -7603,17 +7608,20 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 					metricReporter.addFetchedBytes(expectedBlockSize, this_block.size());
 
 					// Write this_block to storage
-					state int sinceYield = 0;
+					state Standalone<VectorRef<KeyValueRef>> blockData(this_block, this_block.arena());
+					state Key blockEnd = this_block.size() > 0 && this_block.more ? this_block.end()[-1].key : keys.end;
```

Comment:
Is this the point where you are finding that readThrough is not set?  If so, then that actually does look like a bug to me, but I'm not sure why it does not cause a problem.

The range that should be cleared here should either be up to `readThrough` or `keyAfter(this_block.back())`.  Also, `.back()` is a more clear way to express `.end()[-1]` aka get the last one.